test/python/test_bdd_nexus_multipath.py

```diff
@@ -0,0 +1,255 @@
+    return min_cntlid
+
+
+@pytest.fixture(scope="module")
+def min_cntlid_2():
+    """NVMe minimum controller ID for 2nd nexus."""
+    min_cntlid = 60
+    return min_cntlid
+
+
+@pytest.fixture(scope="module")
+def resv_key():
+    """NVMe reservation key."""
+    resv_key = 0xABCDEF0012345678
+    return resv_key
+
+
+@pytest.fixture(scope="module")
+def resv_key_2():
+    """NVMe reservation key for 2nd nexus."""
+    resv_key = 0x1234567890ABCDEF
+    return resv_key
+
+
+@pytest.fixture(scope="module")
+def create_pools(mayastor_mod, pool_config):
+    hdls = mayastor_mod
+
+    cfg = pool_config
+    pools = []
+
+    pools.append(hdls["ms1"].pool_create(cfg.get("name"), cfg.get("uri")))
+
+    pools.append(hdls["ms2"].pool_create(cfg.get("name"), cfg.get("uri")))
+
+    for p in pools:
+        assert p.state == pb.POOL_ONLINE
+    yield pools
+    try:
+        hdls["ms1"].pool_destroy(cfg.get("name"))
+        hdls["ms2"].pool_destroy(cfg.get("name"))
+    except Exception:
+        pass
+
+
+@pytest.fixture(scope="module")
+def create_replica(mayastor_mod, replica_uuid, create_pools):
+    hdls = mayastor_mod
+    pools = create_pools
+    replicas = []
+
+    UUID, size_mb = replica_uuid
+
+    replicas.append(hdls["ms1"].replica_create(pools[0].name, UUID, size_mb))
+    replicas.append(hdls["ms2"].replica_create(pools[0].name, UUID, size_mb))
+
+    yield replicas
+    try:
+        hdls["ms1"].replica_destroy(UUID)
+        hdls["ms2"].replica_destroy(UUID)
+    except Exception as e:
+        logging.debug(e)
+
+
+@given(
+    "a client connected to two controllers to the same namespace",
+    target_fixture="get_nvme_client",
+)
+def get_nvme_client(create_nexus, create_nexus_2):
+    uri = create_nexus
+    uri2 = create_nexus_2
+    dev = nvme_connect(uri)
+    dev2 = nvme_connect(uri2)
+    return dev, dev2
+
+
+@given("both controllers are online")
+def check_controllers_online(get_nvme_client):
+    devs = get_nvme_client
+    assert devs[0] == devs[1], "should have one namespace"
+    desc = nvme_list_subsystems(devs[0])
+    paths = desc["Subsystems"][0]["Paths"]
+    assert len(paths) == 2, "should have 2 paths"
+
+    for p in paths:
+        assert p["State"] == "live"
+
+
+@when("I start Fio")
+def start_io(get_nvme_client):
+    devs = get_nvme_client
+    job = Fio("job1", "randwrite", devs[0]).build()
+    run_cmd(job)
+
+
+@then("I should be able to see IO flowing to one path only")
+def check_io_one_path(mayastor_mod, create_nexus, create_nexus_2):
+    hdls = mayastor_mod
+
+    # default NUMA io_policy has all IO going to the first controller
+    stat = hdls["ms3"].stat_nvme_controllers()
+    assert stat[0].stats.num_write_ops > 1000
+    assert stat[1].stats.num_write_ops > 1000
+
+    stat = hdls["ms0"].stat_nvme_controllers()
+    assert stat[0].stats.num_write_ops == 0
+    assert stat[1].stats.num_write_ops == 0
+
+    uri = create_nexus
+    uri2 = create_nexus_2
+
+    nvme_disconnect(uri)
```

Comment:
Can we move cleanup code into 'create_nexus...' fixtures, straight before 'nexus_destroy()' ? That would assure cleanup step always takes place regardless of assertions in the test. For instance, If the assertion triggers at line 229, then 'nvme_disconnect()' is not reached.

Reply:
Good idea, I've done this.